ApollosProject/apollos-apps
# feat: Push notifications on follow request

## DESCRIPTION

### What does this PR do, or why is it needed?

![2021-03-18 17 04 21](https://user-images.githubusercontent.com/1637694/111697769-59a67b80-880c-11eb-81b1-37c076f34d41.gif)

Companion PR here: https://github.com/ApollosProject/apollos-templates/pull/296

This is a fun one ... 

Adds in a check for the `actions` that OneSignal sends over, and then lets users provide a custom function to fire when an action matches a passed prop. You can see that being used here: 
https://github.com/ApollosProject/apollos-templates/pull/296/files#diff-9b1509d1d4517465bef7cbb0733f3cf179f1e9b555e50041214d2103218846e6R21-R28

Also adds a call to onesignal to send a notification when a user requests to follow another user. That part is straightforward.


### How do I test this PR?

This is what I did.

1. Clone this PR and it's companion. you need to have a database_url and run postgres.
2. Triple check your env variables. You should have an OneSignal api key and id in your api, and a matching one signal id in your client. 
3. Boot android locally. Check in onesignal that your device showed up. If it doesn't, double check the names of your API keys. 
4. Using graphql, send a follow request to whoever is logged into your android sim. You'll have to use the postgres server to find your apollos id. 
5. Accept that follow request in the sim! 

Note! If you view the notification and then dismiss it, or pull down poorly on the top bar and half view the notification, the button goes away. It's part of the android ui, and it's less of a problem on a real device where it's easy to pull the top bar up and down.
